lib/datadog/core/configuration/option.rb

```diff
@@ -191,11 +189,33 @@ def coerce_env_variable(value)
           when :string, NilClass
             value
           else
-            raise ArgumentError,
+            raise InvalidDefinitionError,
               "The option #{@definition.name} is using an unsupported type option for env coercion `#{@definition.type}`"
           end
         end
 
+        def coerce_env_var_int(value)
+          # Strict integer parsing
+          Integer(value)
```

Comment:
I learned (depressingly) the other day that `Integer(value)` actually supports hex and octal literals => e.g. `Integer("010") => 8` and `Integer("0x10") => 16`.

TL;DR the correct form is `Integer(value, 10)` to specifically ask it to parse only as base 10. Now I'll go back to wondering why this isn't the default >_>

Reply:
I think if only `0x10` (and base 10) were allowed, we wouldn't bat an eye, but octals ruin it for everyone.

Reply:
Fixed it!

Reply:
Yeah, it's hard to accidentally have hex, but so easy to accidentally have octal >_>